third_party/move/move-vm/transactional-tests/tests/tests.rs

```diff
@@ -2,7 +2,175 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-pub const TEST_DIR: &str = "tests";
-use move_transactional_test_runner::vm_test_harness::run_test;
+use fail::FailScenario;
+use libtest_mimic::{Arguments, Trial};
+use move_bytecode_verifier::VerifierConfig;
+use move_compiler_v2::logging;
+use move_model::metadata::LanguageVersion;
+use move_transactional_test_runner::{vm_test_harness, vm_test_harness::TestRunConfig};
+use move_vm_runtime::config::VMConfig;
+use once_cell::sync::Lazy;
+use std::{
+    path::{Path, PathBuf},
+    string::ToString,
+    sync::Arc,
+};
+use walkdir::WalkDir;
 
-datatest_stable::harness!(run_test, TEST_DIR, r".*\.(mvir|move)$");
+/// Runs a test, with custom pre- and post-hooks.
+trait Runner: Sync + Send {
+    fn run(&self, path: &Path, name: &str) -> datatest_stable::Result<()> {
+        run(path, get_config_by_name(name))
+    }
+}
+
+/// Implementation of [Runner] with no hooks;
+struct NoHooksRunner;
+
+impl Runner for NoHooksRunner {}
+
+/// Runner used for paranoid mode testing: configures failpoints to skip bytecode verifier.
+struct SkipBytecodeVerifierRunner;
+
+impl Runner for SkipBytecodeVerifierRunner {
+    fn run(&self, path: &Path, name: &str) -> datatest_stable::Result<()> {
+        let scenario = FailScenario::setup();
+        fail::cfg("skip-verification-for-paranoid-tests", "100%return").unwrap();
+        run(path, get_config_by_name(name))?;
+        scenario.teardown();
+        Ok(())
+    }
+}
+
+// TODO: Deduplicate with compiler-v2 transactional tests.
+#[derive(Clone)]
+struct TestConfig {
+    name: &'static str,
+    runner: Arc<dyn Runner>,
+    experiments: &'static [(&'static str, bool)],
+    /// Run the tests with specified language version.
+    language_version: LanguageVersion,
+    /// VM configuration.
+    vm_config: VMConfig,
+    /// Path substrings for tests to include. If empty, all tests are included.
+    include: &'static [&'static str],
+    /// Path substrings for tests to exclude (applied after the include filter).
+    /// If empty, no additional tests are excluded.
+    exclude: &'static [&'static str],
+}
+
+/// Note that any config which has different output for a test directory *must* be added to the
+/// [SEPARATE_BASELINE] array below, so that a special output file `test.foo.exp` will be used for
+/// outputs comparison when running `test.move` or `test.mvir` for config "foo".
+static TEST_CONFIGS: Lazy<Vec<TestConfig>> = Lazy::new(|| {
+    vec![
+        TestConfig {
+            name: "baseline",
+            runner: Arc::new(NoHooksRunner),
+            experiments: &[],
+            language_version: LanguageVersion::latest(),
+            vm_config: VMConfig {
```

Comment:
nit: we could make a function/method for creating the `VMConfig`, instead of repeating it?